File: app/src/main/java/com/faendir/lightning_launcher/multitool/gesture/GestureLibraryDataSource.java
package com.faendir.lightning_launcher.multitool.gesture;

import android.content.Context;
import androidx.annotation.NonNull;

import com.faendir.lightning_launcher.multitool.util.provider.FileDataSource;

import java.io.File;

/**
 * @author F43nd1r
 * @since 06.11.2017
 */

public class GestureLibraryDataSource implements FileDataSource {

    @Override
    public String getPath() {
        return "lib";
    }

    @Override
    public File getFile(@NonNull Context context) {
        return new File(context.getFilesDir(), "gestureLibrary");
    }
}

File: app/src/main/java/com/faendir/lightning_launcher/multitool/gesture/GestureLibraryDataSource.kt
package com.faendir.lightning_launcher.multitool.gesture

import android.content.Context

import com.faendir.lightning_launcher.multitool.util.provider.FileDataSource

import java.io.File

/**
 * @author F43nd1r
 * @since 06.11.2017
 */

class GestureLibraryDataSource : FileDataSource {

    override fun getPath(): String = "lib"

    override fun getFile(context: Context): File = File(context.filesDir, "gestureLibrary")
}
